jac/jaclang/compiler/parser.py

```diff
@@ -639,9 +648,10 @@ def sub_name(self, kid: list[ast.AstNode]) -> ast.SubTag[ast.Name]:
 
             sub_name: COLON NAME
             """
-            if isinstance(kid[1], ast.Name):
+            self.node_idx = 1
```

Comment:
Don't change `self.node_idx` like this, use `self.consume_token(Tok.COLON)` that'll increment the index.